pkg/app/cockroachdb.go

```diff
@@ -0,0 +1,305 @@
+	"k8s.io/client-go/kubernetes"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/helm"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/log"
+	"github.com/kanisterio/kanister/pkg/poll"
+)
+
+const cReadyTimeout = 10 * time.Minute
+
+type CockroachDB struct {
+	name      string
+	namespace string
+	cacrt     string
+	tlscrt    string
+	tlskey    string
+	cli       kubernetes.Interface
+	chart     helm.ChartInfo
+}
+
+// NewCockroachDB Last tested working version "22.1.5"
+func NewCockroachDB(name string) App {
+	return &CockroachDB{
+		name: name,
+		chart: helm.ChartInfo{
+			Release:  appendRandString(name),
+			RepoName: helm.CockroachDBRepoName,
+			RepoURL:  helm.CockroachDBRepoURL,
+			Chart:    "cockroachdb",
+			Values: map[string]string{
+				"image.tag":        "v22.1.5",
+				"image.pullPolicy": "IfNotPresent",
+			},
+		},
+	}
+}
+
+func (c *CockroachDB) Init(context.Context) error {
+	cfg, err := kube.LoadConfig()
+	if err != nil {
+		return err
+	}
+	c.cli, err = kubernetes.NewForConfig(cfg)
+	return err
+}
+
+func (c *CockroachDB) Install(ctx context.Context, namespace string) error {
+	log.Info().Print("Installing cockroachdb cluster helm chart.", field.M{"app": c.name})
+	c.namespace = namespace
+
+	cli, err := helm.NewCliClient()
+	if err != nil {
+		return errors.Wrap(err, "failed to create helm client")
+	}
+
+	if err = cli.AddRepo(ctx, c.chart.RepoName, c.chart.RepoURL); err != nil {
+		return errors.Wrapf(err, "Failed to install helm repo. app=%s repo=%s", c.name, c.chart.RepoName)
+	}
+
+	err = cli.Install(ctx, fmt.Sprintf("%s/%s", c.chart.RepoName, c.chart.Chart), c.chart.Version, c.chart.Release, c.namespace, c.chart.Values, false)
+	return errors.Wrapf(err, "Failed to install helm chart. app=%s chart=%s release=%s", c.name, c.chart.Chart, c.chart.Release)
+}
+
+func (c *CockroachDB) IsReady(ctx context.Context) (bool, error) {
+	log.Info().Print("Waiting for cockroachdb cluster to be ready.", field.M{"app": c.name, "namespace": c.namespace, "release": c.chart.Release})
+	ctx, cancel := context.WithTimeout(ctx, cReadyTimeout)
+	defer cancel()
+	err := kube.WaitOnStatefulSetReady(ctx, c.cli, c.namespace, c.chart.Release)
+	if err != nil {
+		log.WithError(err).Print("Error Occurred --> ", field.M{"error": err.Error()})
+		return false, err
+	}
+	log.Info().Print("Application instance is ready.", field.M{"app": c.name})
+
+	// Read cluster creds from Secret
+	secret, err := c.cli.CoreV1().Secrets(c.namespace).Get(ctx, fmt.Sprintf("%s-client-secret", c.chart.Release), metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	if _, exist := secret.Data["ca.crt"]; exist {
+		c.cacrt = string(secret.Data["ca.crt"])
+	} else {
+		return false, errors.Errorf("Error: ca.crt not found in the cluster credential %s-client-secret", c.chart.Release)
+	}
```

Comment:
```suggestion
	if _, exist := secret.Data["ca.crt"]; !exist {
	        		return false, errors.Errorf("Error: ca.crt not found in the cluster credential %s-client-secret", c.chart.Release)
	} 
        c.cacrt = string(secret.Data["ca.crt"])
```
what do you think about this?

Reply:
Looks Good ! made the changes